app/src/main/java/org/jellyfin/androidtv/startup/StartupActivity.java

```diff
@@ -181,7 +181,8 @@ public void onResponse(ConnectionResult response) {
                     }
 
                     // Check the server version
-                    if (!AuthenticationHelper.isSupportedServerVersion(response.getServers().get(0))) {
+                    if (!response.getServers().isEmpty() &&
+                            !AuthenticationHelper.isSupportedServerVersion(response.getServers().get(0))) {
```

Comment:
Wouldn't this work better in the section above so users get an error message?

Reply:
The `Unavailable` check above should be triggered before it reaches here. The other case was the one causing the `Exception` because that check was happening after the version check. It's still a good idea to make sure the list isn't empty before trying to grab the first value imo.